### PR TITLE
[perms] Simplify guard operations

### PR DESCRIPTION
**Pull request by easyCZ**
## Related Issue(s)

Fixes #

#### Preview status

gitpod:summary

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`
</details>

/hold

### CODE REVIEW

File: components/server/src/authorization/perms.ts [hunk 1]
```diff
@@ -15,34 +15,6 @@ import {
 } from "../prometheus-metrics";
 import { SpiceDBClient } from "./spicedb";
 
-export type OrganizationOperation =
```
Comment (easyCZ) [inline components/server/src/authorization/perms.ts:18]: Instead, we use the OrganizationPermission type directly

File: components/server/src/workspace/gitpod-server-impl.ts [hunk 2]
```diff
@@ -2714,7 +2715,7 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
     protected async guardTeamOperation(
         teamId: string,
         op: ResourceAccessOp,
-        fineGrainedOp: OrganizationOperation,
+        fineGrainedOp: OrganizationPermission | "not_implemented",
```
Comment (easyCZ) [inline components/server/src/workspace/gitpod-server-impl.ts:2718]: We broaden the type here to allow for explicit "not_implemented"

**Review by svenefftinge — APPROVED** [review]

**Comment by easyCZ** [comment]
It's just constants really, don't think we really need to

**Pull request merged**